formats/polus-tabular-to-feather-plugin/src/main.py

```diff
@@ -0,0 +1,201 @@
+
+    logger.info('BINARY FILE: Scanning directory for binary file pattern... ')
+    if filePattern in binary_patterns:
+        #convert hdf5 to vaex df
+        df = vaex.open(file)
+        return df
+    else:
+        raise FileNotFoundError('No supported binary file extensions were found in the directory. Please check file directory again.' )
+    
+def fcs_to_feather(file,outDir):
+    """Convert fcs file to csv. Copied from polus-fcs-to-csv-converter plugin.
+            
+        Args:
+            file (str): Path to the directory containing the fcs file.
+            outDir (str): Path to save the output csv file.
+                
+        Returns:
+            Converted csv file.
+                
+    """
+    file_name = Path(file).stem
+    feather_filename = file_name + ".feather"
+    logger.info('FCS CONVERSION : Begin parsing data out of .fcs file' + file_name)
+    
+    #Use fcsparser to parse data into python dataframe
+    meta, data = fcsparser.parse(file, meta_data_only=False, reformat_meta=True)
+    
+    #Export the fcs data to vaex df
+    logger.info('FCS CONVERSION: converting data to vaex dataframe...')
+    df = vaex.from_pandas(data)
+    logger.info('writing file...')
+    os.chdir(outDir)
+    logger.info('DF to Feather: Writing Vaex Dataframe to Feather File Format for:' + file_name)
+    df.export_feather(feather_filename,outDir)
+    
+def df_to_feather(input_file,filePattern,outDir):
+    """Convert vaex dataframe to Arrow feather file.
+                
+        Args:
+            inpDir (str): Path to the directory to grab file.
+            filepattern (str): File extension.
+            outDir (str): Path to the directory to save feather file.
+                    
+        Returns:
+            Feather format file.
+                    
+    """  
+    file_name = Path(input_file).stem
+    output_file = file_name + ".feather"
+    logger.info('DF to Feather: Scanning input directory files... ')
+    if filePattern == ".csv":
+        #convert csv to vaex df or hdf5
+        os.chdir(outDir)
+        df = csv_to_df(input_file)    
+    else:
+        df = binary_to_df(input_file,filePattern)  
+        
+    logger.info('writing file...')
+    os.chdir(outDir)
+    logger.info('DF to Feather: Writing Vaex Dataframe to Feather File Format for:' + file_name)
+    df.export_feather(output_file,outDir)
+    
+def remove_files(outDir):
+    logger.info('Removing intermediate files... ')
+    outP = Path(outDir)
+    for file in outP.iterdir():
+        ext = Path(file).suffix
+        if not ext == '.feather':
+            os.remove(file)
+    logger.info('Done')
+    
+def main(inpDir: Path,
+         filePattern: str,
+            outDir: Path,
+            ) -> None:
+        """ Main execution function
+        
+        """
+        
+        if filePattern is None:
+            filePattern = '.*'
+        
+        input_dir = Path(inpDir)
```

Comment:
If we are already using a filepattern, we should use the `FilePattern` object. It will do the file selection for you. You can replace the `input_dir` and `input_file_list` with the following:

```python
fp = filepattern.FilePattern(inpDir,filePattern)
```

Down below you can replace `for file in input_dir.iterdir()` with:
```python
for files in fp:
    file = files[0]
    {rest of your code goes here}
```